File: Website/src/native/BuildConfig.ts
import pkg from "@Package";
import { Native } from "./Native";

/**
 * BuildConfigs for Android
 */
class BuildConfigClass extends Native {
  public constructor() {
    super(window.__buildconfig__);
  }

  public get BUILD_DATE(): number {
    if (this.isAndroid) {
      return this.interface.BUILD_DATE();
    } else {
      return WEB_BUILD_DATE;
    }
  }

  public get VERSION_NAME(): VersionType {
    if (this.isAndroid) {
      return this.interface.VERSION_NAME();
    } else {
      return pkg.config.version_name as VersionType;
    }
  }

  public get VERSION_CODE(): number {
    if (this.isAndroid) {
      return this.interface.VERSION_CODE();
    } else {
      return pkg.config.version_code;
    }
  }

  public get APPLICATION_ID(): string {
    if (this.isAndroid) {
      return this.interface.APPLICATION_ID();
    } else {
      return pkg.name;
    }
  }
  public get DEBUG(): boolean {
    if (this.isAndroid) {
      return this.interface.DEBUG;
    } else {
      return __webpack__mode__ === "development";
    }
  }

  public get BUILD_TYPE(): string {
    if (this.isAndroid) {
      return this.interface.BUILD_TYPE;
    } else {
      return __webpack__mode__;
    }
  }
}

const BuildConfig: BuildConfigClass = new BuildConfigClass();
export { BuildConfig, BuildConfigClass };

File: Android/app/src/main/java/com/dergoogler/core/BuildConfig.java
package com.dergoogler.core;

import android.os.Build;
import android.webkit.JavascriptInterface;

import com.dergoogler.mmrl.BuildConfig;

public class BuildConfig {
    @JavascriptInterface
    public int VERSION_CODE() {
        return com.dergoogler.mmrl.BuildConfig.VERSION_CODE;
    }

    @JavascriptInterface
    public String VERSION_NAME() {
        return com.dergoogler.mmrl.BuildConfig.VERSION_NAME;
    }

    @JavascriptInterface
    public String APPLICATION_ID() {
        return com.dergoogler.mmrl.BuildConfig.APPLICATION_ID;
    }

    @JavascriptInterface
    public int SDK_INT() {
        return Build.VERSION.SDK_INT;
    }

    @JavascriptInterface
    public boolean DEBUG() {
        return com.dergoogler.mmrl.BuildConfig.DEBUG;
    }

    @JavascriptInterface
    public String BUILD_TYPE() {
        return com.dergoogler.mmrl.BuildConfig.BUILD_TYPE;
    }
}
